src/app/application.cpp

```diff
@@ -242,6 +242,48 @@ void Application::processMessage(const QString &message)
         m_paramsQueue.append(params);
 }
 
+void Application::runExternalProgram(BitTorrent::TorrentHandle *const torrent) const
+{
+    QString program = Preferences::instance()->getAutoRunProgram();
+    program.replace("%N", torrent->name());
+    program.replace("%L", torrent->category());
+    program.replace("%F", Utils::Fs::toNativePath(torrent->contentPath()));
+    program.replace("%R", Utils::Fs::toNativePath(torrent->rootPath()));
+    program.replace("%D", Utils::Fs::toNativePath(torrent->savePath()));
+    program.replace("%C", QString::number(torrent->filesCount()));
+    program.replace("%Z", QString::number(torrent->totalSize()));
+    program.replace("%T", torrent->currentTracker());
+    program.replace("%I", torrent->hash());
+
+    Logger *logger = Logger::instance();
+    logger->addMessage(tr("Torrent: %1, running external program, command: %2").arg(torrent->name()).arg(program));
+
+#if defined(Q_OS_UNIX)
+    QProcess::startDetached(QLatin1String("/bin/sh"), {QLatin1String("-c"), program});
```

Comment:
Just a second, please!
@Chocobo1: consider using `execvp()`, which looks up the executable in PATH, please.